src/select.js

```diff
@@ -167,12 +167,11 @@ const CanopySelect = React.createClass({
 		var i = this.getIndexFromString(searchString);
 
 		if(i > -1) {
-			this.setState({
-				selectedIndex: i,
-				//			dialogDisplayed: true
-			});
+			this.selectItem(i);
 		}
 
+		clearTimeout(keyTimeout);
```

Comment:
If the timeout isn't cleared each time we come into this method, then the search string will always reset after 1 second, instead we want them to be able to keep typing. Only reset if they stop typing for longer than a second in between each key.